<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from sentinel.urdf.xacro            | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="sentinel">
  <!-- Define materials -->
  <material name="NavyBlue">
    <color rgba="0.0 0.0 0.5 1.0"/>
  </material>
  <material name="WoodYellow">
    <color rgba="1.0 0.8 0.3 0.5"/>
  </material>
  <material name="BlackWheel">
    <color rgba="0.0 0.0 0.0 1.0"/>
  </material>
  <material name="Red">
    <color rgba="0.1 0.0 0.0 1.0"/>
  </material>
  <material name="White">
    <color rgba="0.9 0.9 0.9 1.0"/>
  </material>
  <material name="AbsWhite">
    <color rgba="1 1 1 1.0"/>
  </material>
  <material name="Grey">
    <color rgba="0.7 0.7 0.7 1.0"/>
  </material>
  <material name="DarkGold">
    <color rgba="0.8 0.6 0.2 1.0"/>
  </material>
  <material name="DarkGreen">
    <color rgba="0.0 0.3 0.0 1.0"/>
  </material>
  <link name="base_link">
    <visual>
      <geometry>
        <!--- boy en yukseklik !-->
        <box size="0.245 0.15 0.005"/>
      </geometry>
      <origin rpy="0.0 0.0 0.0" xyz="0 0 0.04"/>
      <material name="WoodYellow"/>
    </visual>
  </link>
  <link name="odom"/>

  <link name="mid_link">
    <visual>
      <geometry>
        <!--- boy en yukseklik !-->
        <box size="0.245 0.15 0.005"/>
      </geometry>
      <origin rpy="0.0 0.0 0.0" xyz="0 0 0.04"/>
      <material name="WoodYellow"/>
    </visual>
  </link>
  <link name="top_link">
    <visual>
      <geometry>
        <!--- boy en yukseklik !-->
        <box size="0.245 0.15 0.005"/>
      </geometry>
      <origin rpy="0.0 0.0 0.0" xyz="0 0 0.04"/>
      <material name="WoodYellow"/>
    </visual>
  </link>
  <!-- PowerBank -->
  <link name="powerbank">
    <visual>
      <geometry>
        <box size="0.15 0.09 0.015"/>
      </geometry>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <material name="AbsWhite"/>
    </visual>
  </link>
  <!-- Raspberry -->
  <link name="raspberry">
    <visual>
      <geometry>
        <box size="0.085 0.056 0.011"/>
      </geometry>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <material name="DarkGreen"/>
    </visual>
  </link>
  <!-- Lidar !-->
  <link name="lidar">
    <visual>
      <geometry>
        <cylinder length="0.02" radius="0.03"/>
      </geometry>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <material name="NavyBlue"/>
    </visual>
  </link>
  <!-- Camera !-->
  <link name="camera">
    <visual>
      <geometry>
        <box size="0.025 0.025 0.01"/>
      </geometry>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <material name="Red"/>
    </visual>
  </link>
  <link name="usb_left">
    <visual>
      <geometry>
        <box size="0.01 0.01 0.01"/>
      </geometry>
      <origin rpy="0.0 0.0 0.0" xyz="0.00 0.00 0.00"/>
      <material name="White"/>
    </visual>
    <visual>
      <geometry>
        <box size="0.004 0.009 0.004"/>
      </geometry>
      <origin rpy="0.0 0.0 0.0" xyz="0.004 0.00 0.0025"/>
      <material name="Grey"/>
    </visual>
    <visual>
      <geometry>
        <box size="0.004 0.009 0.004"/>
      </geometry>
      <origin rpy="0.0 0.0 0.0" xyz="0.004 0.00 -0.0025"/>
      <material name="Grey"/>
    </visual>
  </link>
  <link name="usb_right">
    <visual>
      <geometry>
        <box size="0.01 0.01 0.01"/>
      </geometry>
      <origin rpy="0.0 0.0 0.0" xyz="0.00 0.00 0.00"/>
      <material name="White"/>
    </visual>
    <visual>
      <geometry>
        <box size="0.004 0.009 0.004"/>
      </geometry>
      <origin rpy="0.0 0.0 0.0" xyz="0.004 0.00 0.0025"/>
      <material name="Grey"/>
    </visual>
    <visual>
      <geometry>
        <box size="0.004 0.009 0.004"/>
      </geometry>
      <origin rpy="0.0 0.0 0.0" xyz="0.004 0.00 -0.0025"/>
      <material name="Grey"/>
    </visual>
  </link>
  <link name="left_front_wheel">
    <visual>
      <geometry>
        <cylinder length="0.026" radius="0.0325"/>
      </geometry>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <material name="BlackWheel"/>
    </visual>
  </link>
  <link name="left_back_wheel">
    <visual>
      <geometry>
        <cylinder length="0.026" radius="0.0325"/>
      </geometry>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <material name="BlackWheel"/>
    </visual>
  </link>
  <link name="right_front_wheel">
    <visual>
      <geometry>
        <cylinder length="0.026" radius="0.0325"/>
      </geometry>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <material name="BlackWheel"/>
    </visual>
  </link>
  <link name="right_back_wheel">
    <visual>
      <geometry>
        <cylinder length="0.026" radius="0.0325"/>
      </geometry>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <material name="BlackWheel"/>
    </visual>
  </link>
  <link name="base_mid_left_front">
    <visual>
      <geometry>
        <cylinder length="0.020" radius="0.002"/>
      </geometry>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <material name="DarkGold"/>
    </visual>
  </link>
  <link name="base_mid_left_front_2">
    <visual>
      <geometry>
        <cylinder length="0.020" radius="0.002"/>
      </geometry>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <material name="DarkGold"/>
    </visual>
  </link>
  <link name="base_mid_right_front">
    <visual>
      <geometry>
        <cylinder length="0.020" radius="0.002"/>
      </geometry>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <material name="DarkGold"/>
    </visual>
  </link>
  <link name="base_mid_right_front_2">
    <visual>
      <geometry>
        <cylinder length="0.020" radius="0.002"/>
      </geometry>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <material name="DarkGold"/>
    </visual>
  </link>
  <link name="base_mid_left_back">
    <visual>
      <geometry>
        <cylinder length="0.020" radius="0.002"/>
      </geometry>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <material name="DarkGold"/>
    </visual>
  </link>
  <link name="base_mid_left_back_2">
    <visual>
      <geometry>
        <cylinder length="0.020" radius="0.002"/>
      </geometry>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <material name="DarkGold"/>
    </visual>
  </link>
  <link name="base_mid_right_back">
    <visual>
      <geometry>
        <cylinder length="0.020" radius="0.002"/>
      </geometry>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <material name="DarkGold"/>
    </visual>
  </link>
  <link name="base_mid_right_back_2">
    <visual>
      <geometry>
        <cylinder length="0.020" radius="0.002"/>
      </geometry>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <material name="DarkGold"/>
    </visual>
  </link>
  <link name="mid_top_left_front">
    <visual>
      <geometry>
        <cylinder length="0.080" radius="0.002"/>
      </geometry>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <material name="White"/>
    </visual>
  </link>
  <link name="mid_top_left_back">
    <visual>
      <geometry>
        <cylinder length="0.080" radius="0.002"/>
      </geometry>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <material name="White"/>
    </visual>
  </link>
  <link name="mid_top_right_front">
    <visual>
      <geometry>
        <cylinder length="0.080" radius="0.002"/>
      </geometry>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <material name="White"/>
    </visual>
  </link>
  <link name="mid_top_right_back">
    <visual>
      <geometry>
        <cylinder length="0.080" radius="0.002"/>
      </geometry>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <material name="White"/>
    </visual>
  </link>
  <!-- Body base-mid Joint !-->
  <joint name="base_mid_joint" type="fixed">
    <parent link="base_link"/>
    <child link="mid_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.025"/>
  </joint>
  <!-- Base-powerbank joint-->
  <joint name="base_powerbank" type="fixed">
    <parent link="base_link"/>
    <child link="powerbank"/>
    <origin rpy="0.0 0.0 0.0" xyz="-0.015 0.0 0.075"/>
  </joint>
  <!-- Base-raspberry joint-->
  <joint name="base_raspberry" type="fixed">
    <parent link="base_link"/>
    <child link="raspberry"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.01 0.0 0.088"/>
  </joint>
  <!-- Base usb_joint left joint-->
  <joint name="base_usb_left" type="fixed">
    <parent link="base_link"/>
    <child link="usb_left"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.048 -0.02 0.0985"/>
  </joint>
  <!-- Base usb_joint right joint-->
  <joint name="base_usb_right" type="fixed">
    <parent link="base_link"/>
    <child link="usb_right"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.048 -0.006 0.0985"/>
  </joint>
  <!-- Body base-top Joint !-->
  <joint name="base_top_joint" type="fixed">
    <parent link="base_link"/>
    <child link="top_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.00 0.00 0.11"/>
  </joint>
  <!-- Body & Lidar Joint !-->
  <joint name="base_lidar_joint" type="fixed">
    <parent link="base_link"/>
    <child link="lidar"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.04 0.04 0.19"/>
  </joint>
  <!-- Body & Camera Joint !-->
  <joint name="base_camera_joint" type="fixed">
    <parent link="base_link"/>
    <child link="camera"/>
    <origin rpy="1.57 0.0 1.57" xyz="0.1 0.025 0.1"/>
  </joint>
  <!-- Body & Left Front Wheel Joint !-->
  <joint name="base_left_front_wheel_joint" type="continuous">
    <parent link="base_link"/>
    <child link="left_front_wheel"/>
    <origin rpy="1.6 0.0 0.0" xyz="0.0575 0.09 0.048"/>
    <axis xyz="0.0 0.0 -1.0"/>
  </joint>
  <!-- Body & Right Front Wheel Joint !-->
  <joint name="base_right_front_joint" type="continuous">
    <parent link="base_link"/>
    <child link="right_front_wheel"/>
    <origin rpy="1.6 0.0 0.0" xyz="0.0575 -0.09 0.048"/>
    <axis xyz="0.0 0.0 -1.0"/>
  </joint>
  <!-- Body & Right Front Wheel Joint !-->
  <joint name="base_left_back_joint" type="continuous">
    <parent link="base_link"/>
    <child link="left_back_wheel"/>
    <origin rpy="1.6 0.0 0.0" xyz="-0.0575 0.09 0.048"/>
    <axis xyz="0.0 0.0 -1.0"/>
  </joint>
  <!-- Body & Right Front Wheel Joint !-->
  <joint name="base_right_back_joint" type="continuous">
    <parent link="base_link"/>
    <child link="right_back_wheel"/>
    <origin rpy="1.6 0.0 0.0" xyz="-0.0575 -0.09 0.048"/>
    <axis xyz="0.0 0.0 -1.0"/>
  </joint>
  <!-- Cylindder base-mid left front Joint!-->
  <joint name="base_mid_left_joint" type="fixed">
    <parent link="base_link"/>
    <child link="base_mid_left_front"/>
    <origin rpy="0 0.0 0" xyz="0.11 0.05 0.0525"/>
  </joint>
  <!-- Cylindder base-mid left front2 Joint!-->
  <joint name="base_mid_left_joint_2" type="fixed">
    <parent link="base_link"/>
    <child link="base_mid_left_front_2"/>
    <origin rpy="0 0.0 0" xyz="0.09 0.06 0.0525"/>
  </joint>
  <!-- Cylindder base-mid right front Joint!-->
  <joint name="base_mid_right_joint" type="fixed">
    <parent link="base_link"/>
    <child link="base_mid_right_front"/>
    <origin rpy="0 0.0 0" xyz="0.11 -0.05 0.0525"/>
  </joint>
  <!-- Cylindder base-mid right front2 Joint!-->
  <joint name="base_mid_right_joint_2" type="fixed">
    <parent link="base_link"/>
    <child link="base_mid_right_front_2"/>
    <origin rpy="0 0.0 0" xyz="0.09 -0.06 0.0525"/>
  </joint>
  <!-- Cylindder base-mid left back Joint!-->
  <joint name="base_mid_left_back_joint" type="fixed">
    <parent link="base_link"/>
    <child link="base_mid_left_back"/>
    <origin rpy="0 0.0 0" xyz="-0.11 0.05 0.0525"/>
  </joint>
  <!-- Cylindder base-mid left back2 Joint!-->
  <joint name="base_mid_left_back_joint_2" type="fixed">
    <parent link="base_link"/>
    <child link="base_mid_left_back_2"/>
    <origin rpy="0 0.0 0" xyz="-0.09 0.06 0.0525"/>
  </joint>
  <!-- Cylindder base-mid right back Joint!-->
  <joint name="base_mid_right_back_joint" type="fixed">
    <parent link="base_link"/>
    <child link="base_mid_right_back"/>
    <origin rpy="0 0.0 0" xyz="-0.11 -0.05 0.0525"/>
  </joint>
  <!-- Cylindder base-mid right back2 Joint!-->
  <joint name="base_mid_right_back_joint_2" type="fixed">
    <parent link="base_link"/>
    <child link="base_mid_right_back_2"/>
    <origin rpy="0 0.0 0" xyz="-0.09 -0.06 0.0525"/>
  </joint>
  <!-- Cylindder mid-top left front Joint!-->
  <joint name="mid_top_left_front_joint" type="fixed">
    <parent link="base_link"/>
    <child link="mid_top_left_front"/>
    <origin rpy="0 0.0 0" xyz="0.11 0.045 0.1075"/>
  </joint>
  <!-- Cylindder mid-top left front Joint!-->
  <joint name="mid_top_right_front_joint" type="fixed">
    <parent link="base_link"/>
    <child link="mid_top_right_front"/>
    <origin rpy="0 0.0 0" xyz="0.11 -0.045 0.1075"/>
  </joint>
  <!-- Cylindder mid-top left front Joint!-->
  <joint name="mid_top_left_back_joint" type="fixed">
    <parent link="base_link"/>
    <child link="mid_top_left_back"/>
    <origin rpy="0 0.0 0" xyz="-0.11 0.045 0.1075"/>
  </joint>
  <!-- Cylindder mid-top left front Joint!-->
  <joint name="mid_top_right_back_joint" type="fixed">
    <parent link="base_link"/>
    <child link="mid_top_right_back"/>
    <origin rpy="0 0.0 0" xyz="-0.11 -0.045 0.1075"/>
  </joint>
    <!-- Joint to connect odom to base_link (fixed transform) -->
  <joint name="odom_to_base_link" type="fixed">
    <parent link="odom"/>
    <child link="base_link"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
  </joint>
</robot>
